### PR TITLE
Async projection

### PR DESCRIPTION
**Pull request by mojobec**

### CODE REVIEW

File: Chorus.LinqToXrm.Async/AsyncQuery.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using System.Threading;
+using Chorus.LinqToFetch;
+
+namespace Chorus.LinqToXrm.Async
+{
+    public class AsyncQuery<T> : Query, IQueryable<T>, IAsyncEnumerable<T>
+    {
+        private readonly AsyncQueryProvider _provider;
+
+        public AsyncQuery(BaseQueryProvider provider)
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/AsyncQuery.cs]:
```suggestion
        public AsyncQuery(AsyncQueryProvider provider)
```

File: Chorus.LinqToXrm.Async/AsyncQuery.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            : base(provider, typeof(T))
+        {
+            _provider = (AsyncQueryProvider)provider;
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/AsyncQuery.cs]:
```suggestion
            _provider = provider;
```

File: Chorus.LinqToXrm.Async/AsyncQuery.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        }
+
+        public AsyncQuery(BaseQueryProvider provider, Expression expression)
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/AsyncQuery.cs]:
```suggestion
        public AsyncQuery(AsyncQueryProvider provider, Expression expression)
```

File: Chorus.LinqToXrm.Async/AsyncQuery.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            : base(provider, expression, typeof(T))
+        {
+            _provider = (AsyncQueryProvider)provider;
+        }
+
+        public AsyncQuery(BaseQueryProvider provider, string entityName)
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/AsyncQuery.cs]:
```suggestion
        public AsyncQuery(AsyncQueryProvider provider, string entityName)
```

File: Chorus.LinqToXrm.Async/XrmAsyncFetchService.cs [hunk 5]
```diff
@@ -57,31 +57,10 @@ public string GetEntityLogicalName(string element)
...
-            else
-            {
-                authResult =
-                    await authContext.AcquireTokenAsync(ap.Resource, new ClientCredential(_clientId, _clientSecret));
-            }
-
-            _client.DefaultRequestHeaders.Authorization =
-                new AuthenticationHeaderValue(authResult.AccessTokenType, authResult.AccessToken);
-
+            var authContext = new AuthenticationContext(ap.Authority, false);
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/XrmAsyncFetchService.cs]:
```suggestion
            var authContext = new AuthenticationContext(@"https://login.microsoftonline.com/common", false);
```
We had a problem in the past that ap.Authority gave a wrong url.

File: Chorus.LinqToXrm.Async/XrmAsyncFetchService.cs [hunk 6]
```diff
@@ -57,31 +57,10 @@ public string GetEntityLogicalName(string element)
 
         private async Task Auth()
         {
-            // Turn off spitting out token acquisition logs.
-            LoggerCallbackHandler.UseDefaultLogging = false;
             var ap = await AuthenticationParameters.CreateFromResourceUrlAsync(new Uri($"{_url}{ApiPath}"));
-            var authContext = new AuthenticationContext(@"https://login.microsoftonline.com/common", false);
-
-            AuthenticationResult authResult;
-            if (_clientSecret == null)
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/XrmAsyncFetchService.cs]: Why did you remove the option to authenticate with client secret?

File: Chorus.LinqToXrm.Async/AsyncQuery.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading;
+using Chorus.LinqToFetch;
+
+namespace Chorus.LinqToXrm.Async
+{
+    public class AsyncQuery<T> : Query, IQueryable<T>, IAsyncEnumerable<T>
```
Comment (aronmek) [inline Chorus.LinqToXrm.Async/AsyncQuery.cs]: Can we have a single Query<T> that supports both APIs?

Comment (mojobec) [reply Chorus.LinqToXrm.Async/AsyncQuery.cs]: no, because of we

Comment (mojobec) [reply Chorus.LinqToXrm.Async/AsyncQuery.cs]: should be possible, will add to next version
